streamz/collection.py

```diff
@@ -242,9 +246,12 @@ def verify(self, x):
                             (self._subtype, type(x)))
 
 
-def stream_type(example, stream_type='streaming'):
+def get_stream_type(example, stream_type='streaming'):
     for typ, s_type in _stream_types[stream_type]:
-        if isinstance(example, typ):
+        if isinstance(typ, types.FunctionType):
+            if typ(example):
```

Comment:
Comment here that for a set of types, `typ` here is a function that does the check. Should the other `typ`s also be functions, would that be cleaner?

Reply:
I would need to implement functions for all those different types. I think there are more than a few types that I need to handle if I have to do this. Having said that, it would be nice if I can avoid having a separate function for type checking Frame types.